### DISCUSSION TITLE
advise on best way to get reference dataset

### DISCUSSION THREAD

**Discussion by edzob** [danielnsilva/semanticscholar#53] [General]
Hi, I am new to semantic scholar, here in python as new to the service.

I looked at https://api.semanticscholar.org/api-docs/graph#tag/Paper-Data/operation/get_graph_get_paper_citations

I would love to get specific fields as result based a a queriy for a specific article. 

For example: 
https://api.semanticscholar.org/graph/v1/paper/649def34f8be52c8b66281af98ae884c09aef38b/references?offset=0&limit=1000&fields=paperId,corpusId,url,title,publicationVenue,year,authors,externalIds,citationCount

What would be the best way to do this with semanticscholar?

I would think it is a search_paper query but when I search with the semantic scholar identifier I recieve errors. 

Any tips on how to proceed? 

Regards Edzo

**Comment by danielnsilva** [comment]
Despite the substantial delay in my reply, I'll register a solution for future reference..

To retrieve specific fields for a paper from Semantic Scholar, you can use the [`fields` query parameter](https://github.com/danielnsilva/semanticscholar#fields-list):

```python
from semanticscholar import SemanticScholar       
sch = SemanticScholar()
fields = ['paperId','corpusId','url','title','publicationVenue','year','authors','externalIds','citationCount']
results = sch.get_paper_references('649def34f8be52c8b66281af98ae884c09aef38b', fields)
for item in results:
    print(item)
```
